lib/web_ui/lib/src/engine/scene_painting.dart

```diff
@@ -0,0 +1,20 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:ui/ui.dart' as ui;
+
+// These are additional APIs that are not part of the `dart:ui` interface that
+// are needed internally to properly implement a `SceneBuilder` on top of the
+// generic Canvas/Picture api.
+abstract class SceneCanvas implements ui.Canvas {
+  void saveLayerWithFilter(ui.Rect? bounds, ui.Paint paint, ui.ImageFilter filter);
+}
+
+abstract class ScenePicture implements ui.Picture {
+  ui.Rect get cullRect;
```

Comment:
Should this be called `bounds`? I'm not sure how a picture could be "culling" anything 🤔

Reply:
This terminology comes from skia itself, see `SkPicture::cullRect`. I think the idea is that if that rectangle can be culled by another operation (like a clip operation or something), it can cull the picture and skip all of its drawing commands.